calico_node/startup/startup.go

```diff
@@ -787,12 +787,12 @@ func ensureDefaultConfig(cfg *api.CalicoAPIConfig, c *client.Client, node *api.N
 	}
 
 	// Store the Calico Version as a global felix config setting.
-	if err := c.Config().SetFelixConfig("CalicoVersion", "", VERSION); err != nil {
+	if err := checkConflictError(c.Config().SetFelixConfig("CalicoVersion", "", VERSION)); err != nil {
```

Comment:
How about an alternative approach that we check if the value is already set to what we want to set and do nothing if it's already set to that? This kind of conflict can be very common since all the calico-nodes in the  cluster will try to set these values when they boot up. So if we have 100 nodes they will all try to set `CalicoVersion`, `ClusterType` to the **same** value and will likely run into a conflict. 

It's slightly different from ignoring the error when we see the conflict since if we're doing an upgrade on a small cluster and we try to update `CalicoVersion` and see a conflict, we can't just ignore that. Maybe we want to check if the value is what we want to set to, if not then we retry.

WDYGT?

Reply:
Seems ok to me, we're already doing that with some other values via `ensureGlobalFelixConfig` would be trivial to include this as well.

Reply:
That on its own won't be enough.  There's still a race condition between when we check what the value is and when we try to write a new value.

Reply:
Yep, but inside the `ensureGlobalFelixConfig` I've added a check for said error as well, I don't know if or how much more expensive a `SetFelixConfig` is over a `GetFelixConfig` so maybe the impact would but pretty much nil?

Reply:
Ah I re-read @gunjan5 comment, and yea essentially what is currently happening is the API server sees we're trying to update a value tied to an invalid resource version because two Nodes were trying to set the same value at the same time, and the one who lost ends up receiving this error, and terminating the startup script.  Since we can be pretty sure the value for the Global configs was just set by another Node we can safely ignore it.